### DISCUSSION TITLE
How to create a new Django app for a cookiecutter-django based project

### DISCUSSION THREAD

**Discussion by curtmwheeler** [cookiecutter/cookiecutter-django#4339] [Q&A] [answered]
#### Django 4.2.1 with django-cookiecutter 2023.05.09: makemigrations doesn't create migration file for new model 

I'm using `Django 4.2.1` with the latest version of `django-cookiecutter (2023.05.09)` on `MacOS Ventura 13.3.1`, and I'm having trouble getting the `makemigrations` command to create a migration file for a new model I've added to my Django project. 

Here's what I've done thus far:

I added the new model to my `models.py` file in my app directory:
```
from django.db import models

class TestModel(models.Model):
    name = models.CharField(max_length=50)
```
I created an empty migration file for the app with the `makemigrations` command:
```
python manage.py makemigrations myapp --empty
```
I ran the `makemigrations` command again to create a migration for the new model, but it didn't create a new migration file:

```
python manage.py makemigrations
```
I've double-checked that the `TestModel` app is included in my `INSTALLED_APPS` list in my `settings.py` file, and I've checked that there are no syntax errors in my `models.py` file.

I've also tried running the `check` command to see if there are any issues with my code, but it doesn't show any problems:
```
python manage.py check
```
I've also tried running the `makemigrations` command with the `--verbosity 3` option to get more detailed output, but it doesn't show any errors or warnings related to the `TestModel` app.

I've also checked that the migration file has the correct name and is located in the correct directory. The migration file is named `0001_initial.py` and is located in the `myapp/migrations/` directory.

This is what my directory tree looks like:

```
└─ base
   ├─ .dockerignore
   ├─ .editorconfig
   ├─ .envs
   │  ├─ .local
   │  │  ├─ .django
   │  │  └─ .postgres
   │  └─ .production
   │     ├─ .django
   │     └─ .postgres
   ├─ .pre-commit-config.yaml
   ├─ .readthedocs.yml
   ├─ CONTRIBUTORS.txt
   ├─ LICENSE
   ├─ MailHog
   ├─ README.md
   ├─ SECURITY.md
   ├─ base
   │  ├─ __init__.py
   │  ├─ conftest.py
   │  ├─ contrib
   │  │  ├─ __init__.py
   │  │  └─ sites
   │  │     ├─ __init__.py
   │  │     └─ migrations
   │  │        ├─ 0001_initial.py
   │  │        ├─ 0002_alter_domain_unique.py
   │  │        ├─ 0003_set_site_domain_and_name.py
   │  │        ├─ 0004_alter_options_ordering_domain.py
   │  │        ├─ __init__.py
   │  ├─ media
   │  ├─ static
   │  │  ├─ css
   │  │  ├─ fonts
   │  │  ├─ images
   │  │  ├─ js
   │  │  ├─ sass
   │  │  └─ webpack_bundles
   │  ├─ templates
   │  │  ├─ 403.html
   │  │  ├─ 404.html
   │  │  ├─ 500.html
   │  │  ├─ account
   │  │  │  ├─ account_inactive.html
   │  │  │  ├─ base.html
   │  │  │  ├─ email.html
   │  │  │  ├─ email_confirm.html
   │  │  │  ├─ login.html
   │  │  │  ├─ logout.html
   │  │  │  ├─ password_change.html
   │  │  │  ├─ password_reset.html
   │  │  │  ├─ password_reset_done.html
   │  │  │  ├─ password_reset_from_key.html
   │  │  │  ├─ password_reset_from_key_done.html
   │  │  │  ├─ password_set.html
   │  │  │  ├─ signup.html
   │  │  │  ├─ signup_closed.html
   │  │  │  ├─ verification_sent.html
   │  │  │  └─ verified_email_required.html
   │  │  ├─ base.html
   │  │  ├─ pages
   │  │  │  ├─ about.html
   │  │  │  └─ home.html
   │  │  └─ users
   │  │     ├─ user_detail.html
   │  │     └─ user_form.html
   │  ├─ users
   │  │  ├─ __init__.py
   │  │  ├─ adapters.py
   │  │  ├─ admin.py
   │  │  ├─ api
   │  │  ├─ apps.py
   │  │  ├─ context_processors.py
   │  │  ├─ forms.py
   │  │  ├─ managers.py
   │  │  ├─ migrations
   │  │  │  ├─ 0001_initial.py
   │  │  │  ├─ __init__.py
   │  │  ├─ models.py
   │  │  ├─ tasks.py
   │  │  ├─ urls.py
   │  │  └─ views.py
   │  ├─ utils
   │  │  ├─ __init__.py
   │  │  └─ storages.py
   │  └─ website
   │     ├─ __init__.py
   │     ├─ __pycache__
   │     ├─ admin.py
   │     ├─ apps.py
   │     ├─ migrations
   │     │  ├─ 0001_initial.py
   │     │  ├─ __init__.py
   │     ├─ models
   │     │  ├─ __init__.py
   │     ├─ models.py
   │     ├─ tests.py
   │     ├─ urls.py
   │     └─ views.py
   ├─ config
   │  ├─ __init__.py
   │  ├─ api_router.py
   │  ├─ asgi.py
   │  ├─ celery_app.py
   │  ├─ settings
   │  │  ├─ __init__.py
   │  │  ├─ base.py
   │  │  ├─ local.py
   │  │  ├─ production.py
   │  │  └─ test.py
   │  ├─ urls.py
   │  ├─ websocket.py
   │  └─ wsgi.py
   ├─ docs
   ├─ dump.rdb
   ├─ local.yml
   ├─ locale
   │  └─ README.rst
   ├─ manage.py
   ├─ merge_production_dotenvs_in_dotenv.py
   ├─ package-lock.json
   ├─ package.json
   ├─ production.yml
   ├─ pyproject.toml
   ├─ requirements
   ├─ setup.cfg
   ├─ staticfiles
   ├─ tests
   ├─ webpack
   └─ webpack-stats.json
```

What could be causing makemigrations to not create a migration file for my new model? Are there any additional steps I can take to diagnose the problem? Any help would be appreciated!

**Comment by luzfcb** [comment, marked as answer]
Hello @CurtMRosenblad , this is the step by step of creating a new app in a project that was created using `cookiecutter-django`.
Try to use this way and let me know if the step-by-step solves the problem or not.

### <a name="how-to-create-a-new-django-app-for-a-cookiecutter-django-based-project" href="#how-to-create-a-new-django-app-for-a-cookiecutter-django-based-project">How to create a new django app for a cookiecutter-django based project:</a>


1 - create the `<name-of-the-app>` app with `python manage.py startapp <name-of-the-app>`

2 - move `<name-of-the-app>` directory to `<project_slug>` directory

3 - edit `<project_slug>/<name-of-the-app>/apps.py` and 
change `name = "<name-of-the-app>"` to `name = "<project_slug>.<name-of-the-app>"`

4 - add `"<project_slug>.<name-of-the-app>,` on your `LOCAL_APPS` on [config/settings/base.py](https://github.com/cookiecutter/cookiecutter-django/blob/d3d19264d7236dc704273a555e3741b26d0f848a/%7B%7Bcookiecutter.project_slug%7D%7D/config/settings/base.py#L97)

**Reply by abe-101** [reply]
May I suggest this be added to the projects documentation?

**Reply by abe-101** [reply]
https://github.com/cookiecutter/cookiecutter-django/pull/4944
